### PR TITLE
Enhance/neeboo

### PR DESCRIPTION
**Pull request by neeboo**
## Description

added getAddress and ZilAddress class to crypto
added unit tests to getAddress
added blockConfirm to Transaction
added unit tests to Transaction

To enhance with address format transforming and using blockNumber to confirm a transaction

getAddress enables to transform the address format to specific type and limit the input types.
blockConfirm is to trying to implement a better way to get transaction receipt than simple polling.

## Status
- [x] test case passed

### Implementation

- [x] **ready for review**

### CODE REVIEW

File: packages/zilliqa-js-account/src/transaction.ts [hunk 1]
```diff
@@ -192,6 +204,60 @@ export class Transaction implements Signable {
...
+   * Use `RPCMethod.GetLatestBlock` to get latest blockNumber
+   * Use interval to get the latestBlockNumber
+   * After BlockNumber change, then we use `RPCMethod.GetTransaction` to get the receipt
+   *
+   * @param {string} txHash
+   * @param {number} maxblockCount
+   * @param {number} interval interval in milliseconds
+   * @returns {Promise<Transaction>}
+   */
+  async blockConfirm(
```
Comment (iantanwx) [inline packages/zilliqa-js-account/src/transaction.ts]: @neeboo can you rename this to `confirm` and remove the old confirm? It is not a breaking change, and your technique is much friendlier to the seed node's workload. Thanks!

File: packages/zilliqa-js-account/src/transaction.ts [hunk 2]
```diff
@@ -253,7 +319,11 @@ export class Transaction implements Signable {
 
   private setParams(params: TxParams) {
     this.version = params.version;
-    this.toAddr = this.normaliseAddress(params.toAddr);
+    // this.toAddr = this.normaliseAddress(params.toAddr);
```
Comment (iantanwx) [inline packages/zilliqa-js-account/src/transaction.ts]: Please remove comments, thanks!

File: packages/zilliqa-js-crypto/src/util.ts [hunk 3]
```diff
@@ -199,3 +200,87 @@ export const verifyPrivateKey = (privateKey: string): boolean => {
...
+  if (!validation.isString(address)) {
+    throw new Error(`${address} is not string`);
+  }
+  const zilAddr = new ZilAddress(address);
+
+  const validateType: AddressType[] =
+    fromTypes === undefined || fromTypes.length === 0 ? [] : fromTypes;
+  let total = 0;
+
+  total =
```
Comment (iantanwx) [inline packages/zilliqa-js-crypto/src/util.ts]: Is this necessary? I don't think we need `fromTypes`? We just check the input string `address`. I suggest adding `static isValidAddressEncoding(address: string)` to `ZilAddress`. This code is a little bit hard to follow here. Additionally, keep in mind that `enum` types are not visible to JS consumers of this library; it will be hard for them to use `AddressType`.

File: packages/zilliqa-js-crypto/src/zilAddress.ts [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+    const bech32Bool = validation.isBech32(this.raw);
+    const checksumBool = isValidChecksumAddress(this.raw);
+
+    if (addrBool === true && checksumBool === false) {
+      this.addressType = AddressType.bytes20;
+      this.bytes20 = this.raw.startsWith('0x')
+        ? this.raw.substring(2)
+        : this.raw;
+      this.normalize();
+    } else if (addrBool === true && checksumBool === true) {
```
Comment (iantanwx) [inline packages/zilliqa-js-crypto/src/zilAddress.ts]: Please early return instead of using multiple consecutive `if/else` blocks. E.g.

```
if (validation.isAddress(this.raw) { this.normalize(); return; }
if (validation.isBase58(this.raw) { this.normlize(); return; }
...
```

File: packages/zilliqa-js-account/src/transaction.ts (hunk 1 above)
Comment (neeboo) [reply packages/zilliqa-js-account/src/transaction.ts]: Sure , but different mechanism with different params name, like attempts?

File: packages/zilliqa-js-account/src/transaction.ts (hunk 2 above)
Comment (neeboo) [reply packages/zilliqa-js-account/src/transaction.ts]: ok, will do

File: packages/zilliqa-js-crypto/src/zilAddress.ts (hunk 4 above)
Comment (neeboo) [reply packages/zilliqa-js-crypto/src/zilAddress.ts]: Lesson learned! thanks!

File: packages/zilliqa-js-crypto/src/util.ts (hunk 3 above)
Comment (neeboo) [reply packages/zilliqa-js-crypto/src/util.ts]: Try and remake this one

File: packages/zilliqa-js-account/src/transaction.ts (hunk 1 above)
Comment (iantanwx) [reply packages/zilliqa-js-account/src/transaction.ts]: I think that's ok. Right now not many third party developers are using this API. So, we can break it. Anyway, we will soon return `Promise<string>` from `createTransaction` soon.

**Comment by iantanwx** [comment]
Related to #130

**Comment by neeboo** [comment]
Remake PR on next
